### PR TITLE
Add integrations apis

### PR DESCRIPTION
**Pull request by samudra-ajri**
## Overall
API list:
- GET /v1/units
- GET /v1/document-types
- GET /v1/document-classified
- GET /v1/esign/documents/types
- GET /v1/esign/signers
- GET /v1/esign/documents

NB:
- These apis is not implement the auth yet, hence the auth will use keycloack.
- Because the lack of resource, the `GET /v1/esign/documents` has a different body response with the api contract:
The contract:
```
{
  "data": [
        {
          "id": "0001",
          "status": "SUCCESS",
          "file": "http:.../document/12345"
        },
        {
          ...
        }
    ]
 }
```
This api:
```
{
  "data": [
        {
          "id": "0001",
          "status_id": 1,
          "status_name": "SUCCESS"
          "file_url": "http:.../document/12345/file"
        },
        {
          ...
        }
    ]
 }
```

## Related Link:
- API contract: https://s.id/SidebarAPIDocumentation

## Evidence
title: Add integrations apis
project: SIKD
participants: @samudra-ajri @indraprasetya154 @fajarhikmal214

### CODE REVIEW

File: src/app/Http/Controllers/V1/DocumentSignatureFileController.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(Request $request, $id)
+    {
+        $document = DocumentSignature::find($id);
+        $file = $document->getUrlPublicAttribute();
+        if ($document && $file) {
+            $filename = $document->file;
+            $tempFile = tempnam(sys_get_temp_dir(), $filename);
+            copy($file, $tempFile);
```
[the remark this answers is not in the document]
Comment (samudra-ajri) [reply src/app/Http/Controllers/V1/DocumentSignatureFileController.php:26]: This copy function will store the `$file` into a temporary file directory `$tempFile`. This temporary file will be erased after the file has been oppend. 

cmiiw.

File: src/app/Http/Controllers/V1/EsignSignerController.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(Request $request)
+    {
+        $limit = $request->input('limit', 20);
+        $records = People::select('NIP', 'PeopleName as name', 'PeoplePosition as position')
+            ->whereNotNull('NIP')
+            ->where('PeopleIsActive', PeopleIsActiveEnum::ACTIVE())
+            ->paginate($limit);
+        return $records;
```
Comment (fajarhikmal214) [inline src/app/Http/Controllers/V1/EsignSignerController.php:25]: is there no more where clause like `rolecode` ? 'cause all people is just too much

refs. https://gitlab.com/jdsteam/sikd-website/-/blob/develop/application/models/Model_tandatangan.php#L12

Comment (samudra-ajri) [reply src/app/Http/Controllers/V1/EsignSignerController.php:25]: Hho.. I see, so only some people with specific rolecode who can be a signers, isn't it?

Comment (fajarhikmal214) [reply src/app/Http/Controllers/V1/EsignSignerController.php:25]: on the website, yes Sir. So it's like clients have a specific `role code` to represent their unit/organization. And signers list only from that unit/organization

Comment (samudra-ajri) [reply src/app/Http/Controllers/V1/EsignSignerController.php:25]: If I'm not mistaken, the signers is they who has the `is_visible_at_assign_document=true`, isn`t it? @fajarhikmal214

File: src/app/Http/Controllers/V1/DocumentSignatureFileController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (indraprasetya154) [reply src/app/Http/Controllers/V1/DocumentSignatureFileController.php:26]: nice one!

**Review by indraprasetya154 — APPROVED** [review]

**Pull request merged**